### ISSUE TITLE
Spacy Scikit-learn Multithreaded Training Prediction Issue

### ISSUE THREAD

**Issue by azzuwan** [golastmile/rasa_nlu#150]
I'm using the multithreaded-training branch and when I run:
**python -m rasa_nlu.train -c config.json -t 3**
I get this:
 **/usr/local/lib/python2.7/dist-packages/sklearn/metrics/classification.py:1113: UndefinedMetricWarning: F-score is ill-defined and being set to 0.0 in labels with no predicted samples. 'precision', 'predicted', average, warn_for)**
![spacyscikit-multithreaded-problem](https://cloud.githubusercontent.com/assets/816052/22834398/f67364aa-efef-11e6-9771-282dd4a4e02a.png)

**Comment by amn41** [comment]
TL;DR this isn't an error but I would recommend you add more training data. 

The model still trains and runs correctly I presume? The warning you see is actually related to how the training data is used & can also occur with 1 thread. 

Let's say you have 2 intents. Part of the `f-score` is the number of examples correctly predicted to be each intent. If you have very few training examples for, say, `intent_1`, then when your data is split into training and test sets it can happen that your test set doesn't have *any* examples of `intent_1`. That means the `f-score` isn't defined & scikit-learn has to fall back on a default.

**Comment by azzuwan** [comment]
That explains it thoroughly. Thank you Alan.

**Comment by vduckute1812** [comment]
Thank you very much amn41 :+1:

**Comment by ugarciac** [comment]
hello I have the same problem and I have not been able to solve it
I have
training 1277 objects
test 15000 objects
If I use accuracy or f1_micro, it works for me, how can it be solved, help!
